packages/vaex-core/vaex/dataframe_protocol.py

```diff
@@ -0,0 +1,688 @@
+        """
+        if n_chunks == None:
+            size = self.size
+            n_chunks = self.num_chunks()
+            i = self._col.df.evaluate_iterator(self._col, chunk_size=size // n_chunks)
+            iterator = []
+            for i1, i2, chunk in i:
+                iterator.append(_VaexColumn(self._col[i1:i2]))
+            return iterator
+        elif self.num_chunks == 1:
+            size = self.size
+            i = self._col.df.evaluate_iterator(self._col, chunk_size=size // n_chunks)
+            iterator = []
+            for i1, i2, chunk in i:
+                iterator.append(_VaexColumn(self._col[i1:i2]))
+            return iterator
+
+        else:
+            raise ValueError(f"Column {self._col.expression} is already chunked.")
+
+    def get_buffers(self) -> Dict[str, Any]:
+        """
+        Return a dictionary containing the underlying buffers.
+
+        The returned dictionary has the following contents:
+
+            - "data": a two-element tuple whose first element is a buffer
+                      containing the data and whose second element is the data
+                      buffer's associated dtype.
+            - "validity": a two-element tuple whose first element is a buffer
+                          containing mask values indicating missing data and
+                          whose second element is the mask value buffer's
+                          associated dtype. None if the null representation is
+                          not a bit or byte mask.
+            - "offsets": a two-element tuple whose first element is a buffer
+                         containing the offset values for variable-size binary
+                         data (e.g., variable-length strings) and whose second
+                         element is the offsets buffer's associated dtype. None
+                         if the data buffer does not have an associated offsets
+                         buffer.
+        """
+        buffers = {}
+        buffers["data"] = self._get_data_buffer()
+        try:
+            buffers["validity"] = self._get_validity_buffer()
+        except:
+            buffers["validity"] = None
+
+        try:
+            buffers["offsets"] = self._get_offsets_buffer()
+        except:
+            buffers["offsets"] = None
+
+        return buffers
+
+    def _get_data_buffer(self) -> Tuple[_VaexBuffer, Any]:  # Any is for self.dtype tuple
+        """
+        Return the buffer containing the data and the buffer's associated dtype.
+        """
+        _k = _DtypeKind
+        if self.dtype[0] in (_k.INT, _k.UINT, _k.FLOAT, _k.BOOL):
+            # If arrow array is boolean .to_numpy changes values for some reason
+            # For that reason data is transferred to numpy through .tolist
+            if self.dtype[0] == _k.BOOL and isinstance(self._col.values, (pa.Array, pa.ChunkedArray)):
+                buffer = _VaexBuffer(np.array(self._col.tolist(), dtype=bool))
```

Comment:
Not sure why that happens, I'll see if a test fails because of that.

Reply:
Maybe I should restate: the values changed in a strange way when transferring data with `buffer_to_ndarray` in the case of a bool arrow array if `.to_numpy` was used.

But I will try to use previous suggestion  (`.values/.evaluate()`) for arrow and numpy and I will delete the comment if it works.

Reply:
I get the same strange change of values for `evaluate()` also.  It happens in the case of arrow arrays (int and bool) with missing values.

```python
df = vaex.from_arrays(
    arrow_int_m = pa.array([0, 1, 2, None, 0], mask=np.array([0, 0, 0, 1, 1], dtype=bool)),
    arrow_float_m = pa.array([0.5, 1.5, 2.5, None, 0.5], mask=np.array([0, 0, 0, 1, 0], dtype=bool)),
    arrow_bool_m = pa.array([True, False, True, None, True], mask=np.array([0, 0, 1, 1, 0], dtype=bool))
)

col = df.__dataframe__().get_column_by_name('arrow_int_m')
b, d = col.get_buffers()["data"]
buffer_to_ndarray(b, d)
```
output:
```
array([                  0, 4607182418800017408, 4611686018427387904,
         -2251799813685248,   -2251799813685248], dtype=int64)
```

I can commit with the error so you can have a look?